src/python/pants/backend/python/target_types.py

```diff
@@ -586,6 +586,16 @@ class PexIncludeRequirementsField(BoolField):
     )
 
 
+class PexIncludeSourcesField(BoolField):
+    alias = "include_sources"
+    default = True
+    help = softwrap(
+        """
+        Whether to include your first party sources the binary uses in the packaged PEX file.
```

Comment:
I think expanding this doc with why this could be beneficial would make sense. It could be the difference of "what, why!?" to "oh, aha!!" for a novice user.

Reply:
I will say this value mirros `include_requirements` directly, which also doesn't include potential use cases.